### ISSUE TITLE
Velero restoring empty persistent volumes attached with pods on AWS EKS.  

### ISSUE THREAD

**Issue by kartik-prog** [vmware-tanzu/velero#6536]
## We tried backing up a namespace using velero and restoring it in a different namespace in the same aws eks cluster.

**What steps did you take and what happened:**

Install Velero using helm chart.
```
helm install velero vmware-tanzu/velero --version 4.1.3 -f custom-values.yaml -n velero
```

### BackupStorageLocation: AWS S3

<details><summary>NodeAgent</summary>
<p>

nodeAgent:
  podVolumePath: /var/lib/kubelet/pods
  privileged: true
  priorityClassName: ""
  resources:
    requests:
      cpu: 100m
      memory: 100Mi

</p>
</details> 

<details><summary>Init Containers</summary>
<p>

initContainers:
   - name: velero-plugin-for-csi
     image: velero/velero-plugin-for-csi:v0.5.0
     imagePullPolicy: IfNotPresent
     volumeMounts:
       - mountPath: /target
         name: plugins
   - name: velero-plugin-for-aws
     image: velero/velero-plugin-for-aws:v1.7.0
     imagePullPolicy: IfNotPresent
     volumeMounts:
       - mountPath: /target
         name: plugins

</p>
</details> 

## Installed snapshot crds:

```
kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/v6.2.2/client/config/crd/snapshot.storage.k8s.io_volumesnapshotclasses.yaml
kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/v6.2.2/client/config/crd/snapshot.storage.k8s.io_volumesnapshotcontents.yaml
kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/v6.2.2/client/config/crd/snapshot.storage.k8s.io_volumesnapshots.yaml
```

## Commands: 
```
velero backup create <backup-name> --include-namespaces <namespace> --selector "app=<app-name>" --include-cluster-resources=true
velero restore create <restore-name>  --from-backup <backup-name>  --namespace-mappings <old-namespace>:<new-namespace> –include-cluster-resources
```

**What did you expect to happen:**
We expected all the deployments, statefulsets, configs, pvc's, pv's to get restored in another namespace. Everything worked as expected but the pv's restored with the pods were empty.

## Logs:
[backup_describe_pinot-backup-v9.txt](https://github.com/vmware-tanzu/velero/files/12130384/backup_describe_pinot-backup-v9.txt)
[backup_pinot-backup-v9.log](https://github.com/vmware-tanzu/velero/files/12130388/backup_pinot-backup-v9.log)
[restore_pinot-restore-v9.log](https://github.com/vmware-tanzu/velero/files/12130390/restore_pinot-restore-v9.log)
[version.txt](https://github.com/vmware-tanzu/velero/files/12130391/version.txt)
[restore_describe_pinot-restore-v9.txt](https://github.com/vmware-tanzu/velero/files/12130389/restore_describe_pinot-restore-v9.txt)

- Velero version (use `velero version`):

```
Client:
	Version: v1.11.0
	Git commit: -
Server:
	Version: v1.11.0
```
- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"26", GitVersion:"v1.26.0", GitCommit:"b46a3f887ca979b1a5d14fd39cb1af43e7e5d12d", GitTreeState:"clean", BuildDate:"2022-12-08T19:51:43Z", GoVersion:"go1.19.4", Compiler:"gc", Platform:"darwin/arm64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"25+", GitVersion:"v1.25.11-eks-a5565ad", GitCommit:"df109513ba589adaee235b4e658476275d1fde31", GitTreeState:"clean", BuildDate:"2023-06-16T17:35:00Z", GoVersion:"go1.19.10", Compiler:"gc", Platform:"linux/amd64"}
```

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by reasonerjt** [comment]
@kartik-prog 

> Everything worked as expected but the pv's restored with the pods were empty.

Are you talking about the data in these PVCs?

```
restic Backups:
  Completed:
    pinot/pinot-broker-0: config, kube-api-access-5khrm
    pinot/pinot-controller-0: config, data, kube-api-access-6br7f
    pinot/pinot-minion-stateless-6776dbb549-szjxv: config, kube-api-access-88fbk
    pinot/pinot-server-0: config, data, kube-api-access-44qxg
    pinot/pinot-server-1: config, data, kube-api-access-8tpm4
```

Based on the `restore_describe_pinot-restore-v9.txt` the PVCs are restored successfully
```
restic Restores:
  Completed:
    pinot-v2/pinot-broker-0: config
    pinot-v2/pinot-controller-0: config, data
    pinot-v2/pinot-minion-stateless-6776dbb549-szjxv: config
    pinot-v2/pinot-server-0: config, data
    pinot-v2/pinot-server-1: config, data
 ```
 
 Please clarify if these PVs have data when the backup was created but they are empty after the restore.
 
Based on your settings the backup and restore of the data is handled by `node-agent` via Restic, so we need to check the logs on node agents and the status in the CR `podvolumebackup` and `podvolumerestore`.  Please reproduce the issue and use `velero debug` to collect the debug bundle so we can further debug the issue.

**Comment by reasonerjt** [comment]
As we are approaching v1.13 FC and this issue is targeted to be resolved in v1.13
I'm closing it due to inactivity, please open a new issue if you reproduced the problem.
